src/main/scala/mesosphere/marathon/api/v2/AppHelpers.scala

```diff
@@ -43,8 +44,11 @@ object AppHelpers {
     val unreachableStrategy = update
       .unreachableStrategy.map(Raml.fromRaml(_))
       .getOrElse(UnreachableStrategy.default(hasPersistentVols))
+
+    val role = update.role.orNull
```

Comment:
is `orNull` correct? Shouldn't we set the role from the group depending on the config?

EDIT: I had to expand the method. I'm a little confused. What is this method doing?

Reply:
Yeah, I think it is correct, although it looks really weird. 
This method converts an AppUpdate to an App, by converting it to an AppDefinition and than back to an App.... Having a null value in the AppDefinition leads to an Option.empty in the resulting App.
I've added a comment though.